client/src/components/Applications/AppForm/AppFormSchema.js

```diff
@@ -33,33 +33,22 @@ const FormSchema = (app) => {
         })
       );
     } else {
-      field.type = 'text';
-      appFields.schema.appArgs[param.name] = Yup.string();
-      /* TODOv3 https://jira.tacc.utexas.edu/browse/TV3-92
-       *   if (param.semantics.ontology.includes('email')) {
-       *     field.type = 'email';
-       *     appFields.schema.appArgs[param.name] = Yup.string().email(
-       *        'Must be a valid email.'
-       *      );
-       *
-       *  TODOv3 number. was this ever used? previously was:
-       *       appFields.schema.appArgs[param.name] = Yup.number();
-       *       field.type = 'number';
-       *
-       *  TODOV3 agaveFile was:
-       *           field.agaveFile = param.semantics.ontology.includes('agaveFile');
-       */
+      if (p.notes.fieldType === 'email') {
+        appFields.schema.appArgs[param.name] = Yup.string().email(
+          'Must be a valid email.'
+        );
+      } else if (p.notes.fieldType === 'number') {
+        field.type = 'number';
+        appFields.schema.appArgs[param.name] = Yup.number();
+      } else {
+        field.type = 'text';
+        appFields.schema.appArgs[param.name] = Yup.string();
+      }
     }
     if (field.required) {
       appFields.schema.appArgs[param.name] =
         appFields.schema.appArgs[param.name].required('Required');
     }
-    /* TODOv3 https://jira.tacc.utexas.edu/browse/TV3-92
```

Comment:
I didn't add a validator. I know we discussed that in https://docs.google.com/spreadsheets/d/1asrCkdcB1FR_LI-cZj6vxBc4qwpvccDHZUgijz41llE/edit#gid=0 but feels like we need also to define where we place the custom validation message.  I think we ignore this for now or add to a follow-on issue.  @rstijerina , thoughts?

Reply:
I agree I think that's fine. The type validator should be sufficient imo

Reply:
Did we want to consider bool cases?